scripts/coccinelle/unsigned_assignment.cocci

```diff
@@ -0,0 +1,40 @@
+// Find assignments to unsigned variables and add an 'U' to the value
+// Copyright (C) 2018 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+virtual patch
+virtual report
+
+@r_unsigned@
+typedef uint8_t, uint16_t, uint32_t, uint64_t;
+{unsigned char, unsigned short, unsigned int, uint8_t, uint16_t, uint32_t, uint64_t} v;
+constant C;
+position p;
+@@
+
+  v = C@p
+
+@script:python r_rewrite@
+C << r_unsigned.C;
+z;
+@@
+
+if C.isdigit() != True:
+    cocci.include_match(False)
+
+coccinelle.z = C + "U"
+
+@r_subst@
+{unsigned char, unsigned short, unsigned int, uint8_t, uint16_t, uint32_t, uint64_t} r_unsigned.v;
+constant r_unsigned.C;
+identifier r_rewrite.z;
+@@
+
+  v =
+- C
++ z
+
+@script: python depends on report@
+p << r_unsigned.p;
+@@
+print "Unsigned 'U' suffix missing on line %s of %s" % (p[0].line, p[0].file)
```

Comment:
See how report is handled at `scripts/coccinelle/mini_lock.cocci`
and use the same format.

Reply:
Ok.